### PR TITLE
Add documentation about `snap_getBip32Entropy` and `snap_getBip44Entropy`

### PR DESCRIPTION
**Pull request by Mrtenz**
This adds documentation about the new `snap_getBip32Entropy` and `snap_getBip44Entropy` methods, and adds a deprecation notice to the `snap_getBip44Entropy_*` method.

### CODE REVIEW

**Review by JurbinM — DISMISSED** [review]

**Review by gavatron90 — DISMISSED** [review]

File: docs/guide/snaps-rpc-api.md [hunk 1]
```diff
@@ -447,8 +447,275 @@ if (result === true) {
...
+
+::: tab Code
+
+```javascript
+import { SLIP10Node } from '@metamask/key-tree';
+
+// By way of example, we will use Dogecoin, which has a derivation path starting
+// with `m/44'/3'`.
+const dogecoinNode = await wallet.request({
+  method: 'snap_getBip44Entropy',
```
Comment (FrederikBolding) [inline docs/guide/snaps-rpc-api.md:551]:
```suggestion
  method: 'snap_getBip32Entropy',
```

File: docs/guide/snaps-rpc-api.md [hunk 2]
```diff
@@ -447,8 +447,275 @@ if (result === true) {
...
+:::: tabs :options="{ useUrlFragment: false }"
+
+::: tab Manifest
+
+```json
+{
+  "initialPermissions": {
+    "snap_getBip44Entropy": [
+      {
+        "coinType": "3"
```
Comment (FrederikBolding) [inline docs/guide/snaps-rpc-api.md:669]:
```suggestion
        "coinType": 3
```

File: docs/guide/snaps-rpc-api.md [hunk 3]
```diff
@@ -447,8 +447,275 @@ if (result === true) {
...
+
+:::
+
+::: tab Code
+
+```javascript
+import { getBIP44AddressKeyDeriver } from '@metamask/key-tree';
+
+// By way of example, we will use Dogecoin, which has `coin_type` 3.
+const dogecoinNode = await wallet.request({
```
Comment (FrederikBolding) [inline docs/guide/snaps-rpc-api.md:684]:
```
const dogecoinNode = await wallet.request({
  method: 'snap_getBip44Entropy',
  params: 
  {
      coinType: 3,
    },
});
```

File: docs/guide/snaps-rpc-api.md [hunk 4]
```diff
@@ -447,8 +447,275 @@ if (result === true) {
...
+
+```javascript
+import { SLIP10Node } from '@metamask/key-tree';
+
+// By way of example, we will use Dogecoin, which has a derivation path starting
+// with `m/44'/3'`.
+const dogecoinNode = await wallet.request({
+  method: 'snap_getBip44Entropy',
+  params: [
+    {
```
Comment (FrederikBolding) [inline docs/guide/snaps-rpc-api.md:553]: Arrays are not supported by the RPC call currently

**Review by gavatron90 — APPROVED** [review]

**Review by ritave — APPROVED** [review]

**Comment by rekmarks** [comment]
Please ensure that this PR matches the Flask versions that we are releasing (`10.19.x` and `10.20.x`).

**Comment by FrederikBolding** [comment]
This can be merged when we release `10.20.0`

**Pull request merged**